test/batchReceiver.spec.ts

```diff
@@ -213,4 +213,157 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
+
+    while (abandonMsgCount < 10) {
+      receivedMsgs = await queueClient.receiveBatch(1);
+
+      should.equal(receivedMsgs.length, 1);
+      should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+      should.equal(receivedMsgs[0].deliveryCount, abandonMsgCount);
+      abandonMsgCount++;
+
+      await receivedMsgs[0].abandon();
+    }
+
+    await testPeekMsgsLength(queueClient, 0);
+
+    const deadLetterQueuePath = process.env.QUEUE_NAME + "/$DeadLetterQueue";
+    const client = namespace.createQueueClient(deadLetterQueuePath);
+    receivedMsgs = await client.receiveBatch(1);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].complete();
+
+    await testPeekMsgsLength(client, 0);
+    await client.close();
+  });
+
+  it("Abandoning a message 10 times, you get it no more. You can then only get it from the dead letter subscription", async function(): Promise<
+    void
+  > {
+    await topicClient.send(testMessages[0]);
+    let abandonMsgCount = 0;
+    let receivedMsgs;
+    let clientSubscription: SubscriptionClient;
+
+    while (abandonMsgCount < 10) {
+      receivedMsgs = await subscriptionClient.receiveBatch(1);
+
+      should.equal(receivedMsgs.length, 1);
+      should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+      should.equal(receivedMsgs[0].deliveryCount, abandonMsgCount);
+      abandonMsgCount++;
+
+      await receivedMsgs[0].abandon();
+    }
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+
+    const deadLetterTopicPath =
+      process.env.TOPIC_NAME +
+      "/Subscriptions/" +
+      process.env.SUBSCRIPTION_NAME +
+      "/$DeadLetterQueue";
+
+    clientSubscription = namespace.createSubscriptionClient(
+      deadLetterTopicPath ? deadLetterTopicPath : "",
+      process.env.SUBSCRIPTION_NAME ? process.env.SUBSCRIPTION_NAME : ""
+    );
+
+    receivedMsgs = await clientSubscription.receiveBatch(1);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].complete();
+
+    await testPeekMsgsLength(clientSubscription, 0);
+  });
+
+  it("Deferring a message results in not getting the same message again from queue. The message is then gotten using receiveDefferedMessages", async function(): Promise<
+    void
+  > {
+    await queueClient.sendBatch(testMessages);
+    const msgs = await queueClient.receiveBatch(1);
+
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    should.equal(msgs[0].body, testMessages[0].body);
+    should.equal(msgs[0].messageId, testMessages[0].messageId);
+
+    if (msgs[0].sequenceNumber) {
```

Comment:
Instead of nested if/else blocks, we can simplify this code by exiting early. For example:

```
if (!msgs[0].sequenceNumber) {
   // throw the error here
}

// receive msg from queue to test that it is not the same as the deferred msg
// use sequence number to receive the deferred msg

if (!message) {
   // throw error
}

// test message contents
```